### PR TITLE
Added early data (0-RTT) support

### PR DESCRIPTION
**Pull request by moyaldror**
I will later use this commit to test servers for early data support in sslyze.
I tested it using OpenSSL draft22 (the last commit when I built OpenSSL locally was b44a65512a4a0a299f8f817b63df472e74a0007a).

I haven't updated the OpenSSL in this commit because I don't want to break the branch.
If you can do it - it will be great :)
 
I will also add tests for early data in another PR

### CODE REVIEW

**Comment by nabla-c0d3** [comment]
Thanks this is really cool =)

**Comment by moyaldror** [comment]
Glad you liked it! :)
Have you by any chance updated the OpenSSL version? 
Commit 98e1d93454a5d0b34e929f28528756ba9a636e5c added a new API which i think will be great to add.
I will create another PR with the new API, unittests and sample usage

File: nassl/_nassl/nassl_SSL.c [hunk 1]
```diff
@@ -861,6 +901,17 @@ static PyMethodDef nassl_SSL_Object_methods[] =
     {"write", (PyCFunction)nassl_SSL_write, METH_VARARGS,
      "OpenSSL's SSL_write()."
     },
+#ifndef LEGACY_OPENSSL
+    {"write_early_data", (PyCFunction)nassl_SSL_write_early_data, METH_VARARGS,
+     "OpenSSL's SSL_write_ealy_data()."
```
Comment (konklone) [inline nassl/_nassl/nassl_SSL.c:906]: Flagging a typo here - `SSL_write_ealy_data` should be `SSL_write_early_data`. This is still the case in `master`.

Comment (nabla-c0d3) [reply nassl/_nassl/nassl_SSL.c:906]: Thanks

**Pull request merged**